iam-login-service/src/main/java/it/infn/mw/iam/core/oauth/IamOAuth2RequestFactory.java

```diff
@@ -264,10 +268,32 @@ private Map<String, String> updatedTokenRequestParameters(
 
     });
 
+    validateAndUpdateAcrRequest(tokenRequestParameters);
```

Comment:
I don't think this is the right place to request the acr value. We need to request/propagate it during the authentication request (before login), not at the token endpoint.

Reply:
You're right :)